File: src/_001_Stack/ArrayStack.py
class ArrayStack:
    def __init__(self, size):
        self.size = size
        self.stack = [0] * size
        self.top = -1

    def push(self, item):
        self.top += 1
        self.stack[self.top] = item
        print(str(self.stack[self.top]) + " Push!")

    def pop(self):
        print(str(self.stack[self.top]) + " Pop!")
        pop_item = self.stack[self.top]
        self.stack[self.top] = 0
        self.top -= 1
        return pop_item

    def peek(self):
        print(str(self.stack[self.top]) + " Peek!")

# 사용 예시
stack = ArrayStack(5)
stack.push(1)
stack.push(2)
stack.peek()
stack.pop()
stack.peek()

File: src/_001_Stack/ArrayStack.java
package _001_Stack;


public class ArrayStack {
    int top;    //인덱스
    int size;    //스택 배열의 크기
    int [] stack;
    public ArrayStack(int size) {
        this.size = size;
        stack = new int[size];
        top = -1;
    }

    public void push(int item) {
        stack[++top] = item;
        System.out.println(stack[top] + " Push!");
    }
    public void pop() {
        System.out.println(stack[top] + " Pop!");
        int pop = stack[top];
        stack[top--] = 0;
    }
    public void peek() {
        System.out.println(stack[top] + " Peek!");
    }
}
